### PR TITLE
Rewrite array exists to has

### PR DESCRIPTION
**Pull request by taiyang-li**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Rewrite arrayExists(x -> x = 1, arr) -> has(arr, 1), which improve performance by 1.34x

```
select arrayExists(x -> x = 5, materialize(range(10))) from numbers(10000000) format Null; 
0 rows in set. Elapsed: 0.137 sec. Processed 10.02 million rows, 80.18 MB (72.99 million rows/s., 583.92 MB/s.)

select has(materialize(range(10)), 5) from numbers(10000000) format Null; 
0 rows in set. Elapsed: 0.102 sec. Processed 10.02 million rows, 80.18 MB (98.41 million rows/s., 787.28 MB/s.)
```

### CODE REVIEW

File: src/Interpreters/TreeOptimizer.cpp [hunk 1]
```diff
@@ -661,6 +661,7 @@ void optimizeSumIfFunctions(ASTPtr & query)
 
 void optimizeArrayExistsFunctions(ASTPtr & query)
 {
+    std::cout << "start to optimize" << std::endl;
```
Comment (ucasfl) [inline src/Interpreters/TreeOptimizer.cpp:664]: unused code.

**Comment by ucasfl** [comment]
Maybe we also need to implement it for new analyzer.

**Comment by taiyang-li** [comment]
> Maybe we also need to implement it for new analyzer.

it is done.

**Comment by taiyang-li** [comment]
Failed integration tests seems not related to this pr.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**